providers/pool.rb

```diff
@@ -120,120 +120,152 @@ def site_identifier
+    is_new_private_memory = is_new_or_empty_value?(doc.root, "APPPOOL/add/recycling/periodicRestart/@privateMemory", @new_resource.private_mem.to_s)
+    is_new_log_event_on_recycle = is_new_value?(doc.root, "APPPOOL/add/recycling/@logEventOnRecycle", "Time, Requests, Schedule, Memory, IsapiUnhealthy, OnDemand, ConfigChange, PrivateMemory")
 
-    if is_new_log_event_on_recycle
-      was_updated = true
-      cmd = "#{appcmd(node)} set config /section:applicationPools "
-      cmd << "\"/[name='#{@new_resource.pool_name}'].recycling.logEventOnRecycle:"
-      cmd << "PrivateMemory,Memory,Schedule,Requests,Time,ConfigChange,OnDemand,IsapiUnhealthy\""
-      Chef::Log.debug(cmd)
-      shell_out!(cmd)
-      @new_resource.updated_by_last_action(true)
-    end
-    if @new_resource.private_mem && is_new_private_memory
-      was_updated = true
-      cmd = "#{appcmd(node)} set config /section:applicationPools"
-      cmd << " \"/[name='#{@new_resource.pool_name}'].recycling.periodicRestart.privateMemory:"
-      cmd << "#{@new_resource.private_mem}\""
-      Chef::Log.debug(cmd)
-      shell_out!(cmd)
-      @new_resource.updated_by_last_action(true)
-    end
-    if @new_resource.max_proc && is_new_max_processes
-      was_updated = true
-      cmd = "#{appcmd(node)} set apppool \"#{@new_resource.pool_name}\""
-      cmd << " -processModel.maxProcesses:#{@new_resource.max_proc}"
-      Chef::Log.debug(cmd)
-      shell_out!(cmd)
-      @new_resource.updated_by_last_action(true)
-    end
-    if @new_resource.thirty_two_bit && is_new_enable_32_bit_app_on_win_64
-      was_updated = true
-      cmd = "#{appcmd(node)} set apppool \"/apppool.name:#{@new_resource.pool_name}\""
-      cmd << " /enable32BitAppOnWin64:#{@new_resource.thirty_two_bit}"
-      Chef::Log.debug(cmd)
-      shell_out!(cmd)
-      @new_resource.updated_by_last_action(true)
-    end
-    if @new_resource.recycle_after_time && is_new_recycle_after_time
-      was_updated = true
-      cmd = "#{appcmd(node)} set apppool \"/apppool.name:#{@new_resource.pool_name}\""
-      cmd << " /recycling.periodicRestart.time:#{@new_resource.recycle_after_time}"
-      Chef::Log.debug(cmd)
-      shell_out!(cmd)
-      @new_resource.updated_by_last_action(true)
-    end
-    if @new_resource.recycle_at_time && is_new_recycle_at_time
-      was_updated = true
-      cmd = "#{appcmd(node)} set apppool \"/apppool.name:#{@new_resource.pool_name}\""
-      cmd << " /-recycling.periodicRestart.schedule"
-      Chef::Log.debug(cmd)
-      shell_out!(cmd)
-      @new_resource.updated_by_last_action(true)
+    # cpu items
+    is_new_cpu_action = is_new_value?(doc.root, "APPPOOL/add/cpu/@action", @new_resource.cpu_action.to_s) 
+    is_new_cpu_limit = is_new_value?(doc.root, "APPPOOL/add/cpu/@limit", @new_resource.cpu_limit.to_s)
+    is_new_smp_affinitized = is_new_value?(doc.root, "APPPOOL/add/cpu/@smpAffinitized", @new_resource.cpu_smp_affinitized.to_s)
```

Comment:
This is_new_smp_affinitized is referenced as is_new_cup_smp_affinitized later in code.